### ISSUE TITLE
some character names are corrupted

### ISSUE THREAD

**Issue by xangregg** [BirkoRuzicka/Star-Trek-Transcripts#1]
In some cases, the character name key in the JSON file appears to be corrupted by the subsequent dialog. A few of the shorter examples are:

- "PIKE: Vina?\r ORION": ["Vina?\r ORION: Glistening green ...
- "SPOCK: Ready, Captain.": ["Ready, Captain. ...
- "KIRK: But we don't hate you. KUKULKAN": ["But we ...

And sometimes an extra colon or a semicolon appears with the character name:
- "PIKE;": ["I can't help ...
- "SHAHNA:": ["Step away from the door. ...
- "PICARD:": ["Go ahead, Lieutenant." ...

The following regex, looking for a lack of a trailing quote before the colon find 106 occurrences:

>    "[A-Z '0-9]+:

This one find 62 occurrences of a trailing semicolon:

>    "[A-Z '0-9]+;":

I don't know Python well enough to tell if there is an issue in the source or in the scraping or in the cleaning.

**Comment by xangregg** [comment]
At the risk of lumping in a different issue, some character names also contain qualifiers with a "[". Such as:
-  "KOLANA [OC}": ["We are approaching Trelka Five."]
- "GRAT [on viewscreen": ["Check your sensors

**Comment by xangregg** [comment]
I think most of these issues are from the source, and I don't know if your script can fix them readily since they follow a few different patterns. I have a copy of the JSON file with manual to all the issues I could find. Not sure if you want a pull request since it breaks your reproducibility and the diff won't work well with a one-line JSON file. In the mean time, my version is at [xangregg/data/StarTrekDialogue2.json](https://github.com/xangregg/data/blob/master/StarTrekDialogue2.json)

**Comment by BirkoRuzicka** [comment]
Hi! The webscraping script does not take care of this kind of data cleaning, as I'm sure you're aware. The source is littered with these kind of errors (although DISCOVERY seems particularly bad), and my script was not designed to handle the kind of errors you describe. 

I realize what a pain in the nacelles this can be, though, is someone wants to just quickly do some analysis with the data. So I will soon upload an improved version of the JSON, with much fewer of these errors. Hope that will be helpful to other people. 

As you correctly point out, this will break reproducibility - but maybe it's useful nevertheless.

**Comment by xangregg** [comment]
After looking at some of the sources, it's amazing you got as much as you did. For DIS I did notice Burnham as Burham, Burnam, Burnha, and Burnhan!

**Comment by BirkoRuzicka** [comment]
Agreed. I admit I did not check this very thoroughly. At this point I almost expect this to affect the results in general, but from what I saw so far, it would only affirm the trend I found: Countless lines were not attributed correctly to male characters in TOS and TAS, while (as you mentioned) countless lines were not attributed to Burnham in DIS. I'll run the numbers again when I've finished cleaning the data.
